tests/functionaltests/Tests/NSURLConnection.mm

```diff
@@ -23,36 +23,43 @@ typedef NS_ENUM(NSInteger, NSURLConnectionDelegateType) {
     NSURLConnectionDelegateDidFailWithError
 };
 
+// Abstract wrapper around a NSURLConnection that signals a condition when a callback is received.
+// Do not use this class directly - instantiate one of the subclasses instead
 @interface NSURLConnectionTestHelper : NSObject <NSURLConnectionDelegate>
-@property NSCondition* condition;
-@property dispatch_queue_t queue;
-@property NSMutableArray* delegateCallOrder;
+@property (retain) NSCondition* condition;
+@property (assign) dispatch_queue_t queue;
+@property (retain) NSMutableArray* delegateCallOrder;
 @property (copy) NSURLResponse* response;
 @property (copy) NSData* data;
 @property (copy) NSError* error;
 
 - (instancetype)init;
-- (NSURLConnection*)createConnectionWithRequest:(NSString*)URLString;
+- (NSURLConnection*)_createConnectionWithRequest:(NSString*)URLString;
+- (NSURLConnection*)createAndStartConnectionWithRequest:(NSString*)URLString;
 
 @end
 
 @implementation NSURLConnectionTestHelper
-
 - (instancetype)init {
     if (self = [super init]) {
-        self->_condition = [[NSCondition alloc] init];
+        self->_condition = [NSCondition new];
```

Comment:
I generally default to using self-> in inits, to guard against [super init] doing a replacement. That's not really possible here since the superclass is NSObject, so I'll change it, but I disagree that there needs to be a specific reason to use self->, especially since inheritance interactions aren't always this clear...

---
In reply to: [111229643](https://github.com/Microsoft/WinObjC/pull/2464#discussion_r111229643) [](ancestors = 111229643,111229286,111228615)